notebooks/time-to-merge-prediction/thoth-station/thoth_station_time_to_merge_model.ipynb

```diff
@@ -0,0 +1,3063 @@
+{
```

Comment:
I noticed the same behavior in our openshift origin notebooks. I'm not entirely sure. Looking into it

Reply:
Okay so here are my thoughts @chauhankaranraj:

I suspect what's happening here is that the final vector that we send the model is different because although both the datasets have the same set of PRs, however the feature transformed dataset is different.

- In the first case, while sanity checking the trained model, we pass it `X_test` which is a dataset generated by feature transforming the entire datafarme and then splitting in.

- Whats happening at the end of the notebook however is that, we pick rows of the dataframe which match with X_test and the model inference pipeline performs feature transformation only on those rows.

I think whats causing the difference would be the `NumPrevPRsTransformer` column, which depends on the whole dataset, since there is a difference in the "Number of previously merged PRs" in the dataframe depending on how many rows we pass it, i think that column is being transformed differently.

Explained my observation here: https://github.com/aicoe-aiops/ocp-ci-analysis/issues/509 

I also think that the performance degradation is significant and `NumPrevPRsTransformer` seems like an important column, so I think we should plan whether we want to modify its behavior or exclude it.

_Side note:_ Another qs I have is about the feature scaler. In the sklearn pipeline what is the standard scaler fitted to? I know that during our training, we fit it to our train data and scale the test data accordingly, how is that impacted when its in the pipeline?

Reply:
Good catch! I think this is a tricky problem coz imo it makes more sense to perform feature engineering after the train-test split, but in this case if we split first then we won't be possible to compute NumPrevPRs correctly (at least the way it's being computed rn).

So I'm leaning a bit towards removing it for now and seeing how much the performance drop is. If it turns out that we absolutely must keep this feature, then we can explore ways of doing that in another issue. WDYT?

Reply:
Excluded that column from the features and the results now look consistent. 💯 
The performance is slightly lower but not affected too much. So I guess we can do without this feature for now and explore ways of incorporating that in the future in a separate issue
<img width="602" alt="Screen Shot 2022-07-08 at 1 31 42 PM" src="https://user-images.githubusercontent.com/32435206/178041873-cb788009-55bc-4967-9e31-0f40e6e37d92.png">

Reply:
Great! Yeah I agree, these metrics look reasonable (much better than those for openshift origin ttm models), so we can do without this feature :)